src/wattsi.pas

```diff
@@ -504,9 +505,7 @@    TCrossReferences = record
 
                   Scratch := Default(Rope);
                   Scratch.Append('#');
-                  ExtractedData := Element.GetAttribute('id');
-                  Assert(not ExtractedData.IsEmpty);
```

Comment:
EnsureID is verifying that things within EnsureID are acting as expected.
But here you're outside EnsureID, so you can't trust that EnsureID is correct. Hence the assert.